cpp/order/Steinhardt.cc

```diff
@@ -173,17 +173,42 @@ void Steinhardt::baseCompute(const freud::locality::NeighborList* nlist,
 void Steinhardt::computeAve(const freud::locality::NeighborList* nlist,
                                   const freud::locality::NeighborQuery* points, freud::locality::QueryArgs qargs)
 {
-    std::shared_ptr<freud::locality::NeighborIterator> niter =
-        freud::locality::getNeighborIterator(points, points->getPoints(), m_Np, qargs, nlist);
+    std::shared_ptr<locality::NeighborQuery> nq;
+    std::shared_ptr<locality::NeighborQueryIterator> iter;
+    if (nlist == NULL)
+    {
+        if (dynamic_cast<const locality::RawPoints*>(points))
```

Comment:
Should just be in RawPoints -> query

Reply:
bool made_nlist = false;
nlist = points->query(qargs).toNeighborList()
something like that
Later:
if (made_nlist)
   delete stuff

Reply:
Or just resolve to NeighborList on the Cython side.

Reply:
As I mentioned earlier, we can clean this up further by just implementing an iterator for NeighborLists. Not a per-point iterator, but one that goes over the entire NeighborList and implements the `query` function to return `PerPointIterator`s. Since the NeighborListPerPointIterator has already been implemented, it should be pretty easy to do that. I don't think that's super necessary, so I'm not going to do this in this PR; if we find this use-case popping up in many places, though, we can implement that later and it should be pretty easy.